core/src/util.jl

```diff
@@ -374,12 +374,13 @@ storage: tells ForwardDiff whether this call is for differentiation or not
 function get_level(
     p::Parameters,
     node_id::NodeID,
+    i::Integer,
     t::Number;
     storage::Union{AbstractArray, Number} = 0,
 )::Tuple{Bool, Number}
     (; basin, level_boundary) = p
     if node_id.type == NodeType.Basin
-        _, i = id_index(basin.node_id, node_id)
+        i = iszero(i) ? id_index(basin.node_id, node_id)[2] : i
```

Comment:
If it's not a basin, it's the other node of an edge? This could use some (inline) documentation.